drivers/adc/adc_ads114s0x.c

```diff
@@ -820,20 +830,32 @@ static int ads114s0x_channel_setup(const struct device *dev,
 		}
 	}
 
+	ADS114S0X_REGISTER_VBIAS_VB_LEVEL_SET(vbias, config->vbias_level);
+
+	if ((channel_cfg->vbias_pins & ~GENMASK(7, 0)) != 0) {
```

Comment:
Multiple pages of bit definition macros just to miss the bitmask you need and have it harcoded in the middle of the function is a bit silly, should this be expressed in term of something else?

I think it' be cool if that whole block of hal-like code was refactored out of this driver to be honest.

Reply:
There is a small difference with this one, as it is a check for the input. The GENMASK-macros are used for actual register access. But I agree, having a magic number thrown in somewhere doesn't help the readability. Therefore I introduced two defines for these magic numbers.

Otherwise I do not see how you want to get rid of the HAL-like code. It is a driver for a certain piece of hardware, and there is no HAL available for it. Hence, there will be some defines and macros (to avoid magic numbers). I could offer moving them into a separate header?

Reply:
nah just having more targeted defines just for what you need, like any other drivers, anyway, nothing in the scope of this PR I guess, at least it's cohere with itself